src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py

```diff
@@ -0,0 +1,151 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azure.cli.core.commands.parameters import (get_location_type, get_enum_type, tags_type)
+from azure.cli.command_modules.ams._completers import get_role_definition_name_completion_list
+
+from azure.mediav3.models import (EncoderNamedPreset, Priority, AssetContainerPermission)
+
+from ._validators import storage_account_id
+
+# pylint: disable=line-too-long
+
+
+def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statements
+    name_arg_type = CLIArgumentType(options_list=('--name', '-n'), metavar='NAME')
+    account_name_arg_type = CLIArgumentType(options_list=('--account-name', '-a'), metavar='ACCOUNT_NAME')
+    storage_account_arg_type = CLIArgumentType(options_list=('--storage-account'), metavar='STORAGE_NAME')
+    password_arg_type = CLIArgumentType(options_list=('--password', '-p'), metavar='PASSWORD_NAME')
+    transform_name_arg_type = CLIArgumentType(options_list=('--transform-name', '-t'), metavar='TRANSFORM_NAME')
+    expiry_arg_type = CLIArgumentType(options_list=('--expiry'), metavar='EXPIRY_TIME')
+
+    with self.argument_context('ams account') as c:
+        c.argument('account_name', name_arg_type, id_part='name',
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', arg_type=tags_type)
+
+    with self.argument_context('ams account create') as c:
+        c.argument('storage_account', storage_account_arg_type,
+                   help='The name or resource ID of the primary storage account to attach to the Azure Media Services account. Blob only accounts are not allowed as primary.',
+                   validator=storage_account_id)
```

Comment:
The typical convention would be to name this `validate_storage_account_id`.  Alternatively, you can make a type callable, in which case this name makes sense.  It would look something like this:
```Python
def storage_account_id(val):  # val is a string
   val = ... # logic to normalize name or ID to an ID string
   return val
```
The syntax is a little cleaner than a validator and then you register it here as:
`type=storage_account_id`

Regardless, you can move this kwarg into the `storage_account_arg_type`.

Reply:
Done, we renamed to validate_storage_account_id.